src/lib/mdns/DiscoveryManager.cpp

```diff
@@ -191,9 +198,21 @@ CHIP_ERROR DiscoveryManager::PublishUnprovisionedDevice(chip::Inet::IPAddressTyp
     entries[1].mDataSize = strnlen(discriminatorBuf, sizeof(vendorProductBuf));
 
 #if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+    // retrieve serial number
+    SuccessOrExit(error =
+                      chip::DeviceLayer::ConfigurationMgr().GetSerialNumber(serialNumber, sizeof(serialNumber), serialNumberSize));
+    serialNumber[serialNumberSize] = '\0';
+
+    // retieve rotation counter
```

Comment:
nit: consider removing these comments as they restate the code: `...GetRotationCounter...` does a `// retrieve rotation counter` (I think there is a typo in retrieve).

Reply:
updated